src/mlir/generator.cc

```diff
@@ -470,28 +491,106 @@ namespace mlir::verona
     // Create basic-blocks, conditionally branch to if/else
     mlir::ValueRange empty{};
     auto ifBB = currentFunc.addBlock();
-    auto elseBB = currentFunc.addBlock();
+    mlir::Block* elseBB = nullptr;
+    if (hasElse(ast))
```

Comment:
Do we have a style rule relating to braces? I have a strong preference for braces around every conditional body.

Reply:
This is a multi-line (single statement) if/else, so I'll add braces.

We discussed it and the conclusion is that it's generally good for most cases, but it can clutter in error treatment cases like:
```
    auto var = foo(bar);
    if (auto err = var.takeError())
        return err;
    ... continue;
```

My approach is to only refrain from braces if there is a single line on both `if` and `else` (if available) statements. 

But it would also be fine to add to *any* `if-with-else` statement and have it only on `single-line-only-if-block` case, which I have a strong preference for.

Reply:
I'd really like to avoid omitting braces on anything with an else clause.  The compiler will warn in ambiguous cases, but the C grammar does not do what humans expect in this case.

Reply:
That was my point above, I agree with that. The only one I still think we should avoid braces is "if with single line statement and no else".

Reply:
PS: You need to look at the new change to see that I have added braces on the block you asked. If you click on the link above the comment, you will see the old change, which is outdated.